File: Arithmetic_HCFInput.cpp
// Arithmetic.cpp : Defines the entry point for the console application.
//

#include "stdafx.h"
#include "FastHCF.h"

#include <iostream>
#include <vector>
#include <string>
#include <sstream>
#include <regex>

using namespace std;

int main() {
    try {
        cout << "\n    Welcome to our demonstration sequels\n";
        cout << "Hope you enjoy (and follow) the lessons.\n\n";

        vector<unsigned> set;

        // Collect input
        cout << "\nWelcome to our Find HCF program.\n";
        cout << "Enter your series of numbers whose HCF you wish to find.\n";
        cout << "\nType 'done' when you are through with entering your numbers.\n";
        cout << "Enter First Number:  ";

        set = {};
        string user_input;
        int user_num;
        regex verify_num("([0-9]+)"); //Regular expression object
        try {
            do {
                // get keyboard input
                getline(cin, user_input);
                // Convert 'user_input' to upper case.
                transform(user_input.begin(), user_input.end(), user_input.begin(), ::toupper);
                // Make sure input is a number
                if (regex_match(user_input, verify_num)) {
                    stringstream(user_input) >> user_num;
                    if (user_num != 0) {
                        set.push_back(user_num);
                        cout << "Enter Next Number:  ";
                    } else {
                        cout << "\nYou cannot enter zero. Repeat that!\nType 'done' when you're finished.\n";
                    }
                } else if (user_input != "DONE") {
                    cout << "\nWrong Input. Repeat that!\nType 'done' when you're finished.\n";
                }
            } while (user_input != "DONE");
        } catch (exception& ex) {
            throw "Bad Input";
        }

        stringstream aux;
        string render_result = "The HCF of ";

        for (int number : set) {
            aux.str("");
            aux << number;
            render_result += aux.str() + "; ";
        }
        render_result += "is: ";

        // use the fast class to create object
        FastHCF h_c_f(set);
        cout << "\n\n" << render_result << h_c_f.getHCF() << "\n";

    } catch (exception& e) {
        cout << "\n" << e.what() << "\n";
    }
    return 0;
}

File: Arithmetic_HCFInput.pl
#!/usr/bin/perl;
use strict;
use warnings;
use FASTHCF qw(getHCF);

# Useful variables
my ($lower_boundary, $upper_boundary, $test_guy, $answer);
my @set;

# Collect input
print "\nWelcome to our Find HCF program.\n";
print "Enter your series of numbers whose HCF you wish to find.\n";
print "\nType 'done' when you are through with entering your numbers.\n";
print "Enter First Number:  ";

@set = ();
my $user_num;

eval {
	do {
		# get keyboard input
		$user_num = <>;
		chomp $user_num;
		# Convert 'user_num' to upper case.
		$user_num = uc $user_num;
		# Make sure input is a number
		if ($user_num =~ /[0-9]+/) {
			if ($user_num != 0) {
				push @set, $user_num;
				print "Enter Next Number:  ";
			} else {
				print "\nYou cannot enter zero. Repeat that!\nType 'done' when you're finished.\n";
			}
		} elsif ($user_num ne "DONE") {
			print "\nWrong Input. Repeat that!\nType 'done' when you're finished.\n";
		}
	} while ($user_num ne "DONE");
};

# Use the fast HCF module/class
my $h_c_f = FASTHCF->new(\@set);
$answer = $h_c_f->getHCF();
print ("The H.C.F. of ", join(", ", @set), " is $answer\n");


print "\n\n";